### DISCUSSION TITLE
Unable to create re-useable forms using TS

### DISCUSSION THREAD

**Discussion by ChrisSmith-0** [react-hook-form/react-hook-form#7804] [Q&A] [answered]
Describe
Having a lot of trouble while trying to make re-usable controlled form modules while using typescript. My specific use case at work is having three date fields which will be used as a component in multiple different forms. When trying to pass a generic type down into the component to Controller component seems to only want the FieldValues type. When trying the other method of a specific Control<T> I also seem to run into type safety issues. Is anyone able to suggest how this is possible?

To Reproduce
Go to https://codesandbox.io/s/react-hook-form-control-forked-pme5j?file=/src/App.tsx

Codesandbox link
https://codesandbox.io/s/react-hook-form-control-forked-pme5j?file=/src/App.tsx

Screenshots
![image](https://user-images.githubusercontent.com/2602263/153679786-aea69d8f-b1e7-4e2a-87ef-f2e1f57759e6.png)

**Comment by Moshyfawn** [comment, marked as answer]
It might be a current limitation of RHF ATM.
Type-safe reusable components TS issue will be addressed in the next major V8 release of RHF.
For more context, see #7354

**Reply by bluebill1049** [reply]
yes, we are making improvement on v8, name type shouldn't be `string` instead of `FieldPath<T>`

https://react-hook-form.com/ts#FieldPath
